src/opds_dumper.cpp

```diff
@@ -206,10 +239,29 @@ std::string OPDSDumper::categoriesOPDSFeed() const
   );
 }
 
+std::once_flag fillLanguagesFlag;
+
+void fillLanguagesMap() 
+{
+  std::call_once(fillLanguagesFlag, []() {
```

Comment:
I think it is better to call `fillLanguagesMap()` via `std::call_once()` instead of wrapping its entire body in `std::call_once()`. Such a point of view is supported by the function's name and the fact that the languages map object (`iso639_3`) is publicly accessible (i.e. can be cleared or otherwise modified by anyone). Therefore, the request to "fill the languages map" must be performed by the respective function unconditionally. However, for the current code logic, that that request must be made only once.

BTW, why did you choose the solution based on `std::call_once`? An alternative was to make `iso639_3` a static variable inside `getLanguageSelfName()` that is initialized from the return value of a function `getLanguagesMap()`:

```c++
std::map<std::string, std::string> getLanguagesMap() {
  std::map<std::string, std::string> langMap = {"atj","atikamekw"},
    ...
  }
  // add libicu data to langMap
  ...
  return langMap;
}

std::string getLanguageSelfName(const std::string& lang) {
  static const auto iso639_3 = getLanguagesMap();
  ...
}
```

Reply:
> BTW, why did you choose the solution based on std::call_once?

Because I didn't fully get `Singleton` and how it could be used :)

>  An alternative was to make iso639_3 a static variable inside getLanguageSelfName() that is initialized from the return value of a function getLanguagesMap():

This is neat, which one should I prefer?

Reply:
Whichever you like more for this PR

Reply:
I mean it, really. Still advising to explain your choice in a written form. It will help to shape your own taste in software engineering.

Reply:
Okay then, thanks for the freedom!
I searched up differences about the ways and they look fully identical, so purely `call_once` is written *call once* (more intuitve) I will go by it